plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestPinotConfig.java

```diff
@@ -97,4 +102,23 @@ public void testInvalidCountDistinctPushdown()
                 .isInstanceOf(IllegalStateException.class)
                 .hasMessage("Invalid configuration: pinot.aggregation-pushdown.enabled must be enabled if pinot.count-distinct-pushdown.enabled");
     }
+
+    @Test
+    public void testControllerUrls()
+    {
+        assertTrue(isControllerUrlValid("http://my-controller-1:9000,http://my-controller-2:9000", false));
+        assertTrue(isControllerUrlValid("https://my-controller-1:8443,https://my-controller-2:8443", true));
+        assertFalse(isControllerUrlValid("http://my-controller-1:9000,http://my-controller-2:9000", true));
+        assertFalse(isControllerUrlValid("https://my-controller-1:8443,https://my-controller-2:8443", false));
```

Comment:
true. I will not check `isControllerUrlValidWhenTlsDisabled` for now then, so the default behavior is not impacted.

Reply:
Just removing `isControllerUrlValidWhenTlsDisabled` doesn't make sense to me. What do you think about following?  

> I think we should control pinot.controller-urls by this flag only when users omitted scheme

Reply:
Current `pinot.controller-urls` directly parse string to URI, so it gives default scheme HTTP if empty.
If we want to change it, there are two ways:
1. For backward compatibility:
Since user only set `pinot.controller-urls`, whenever we detect HTTPS, we override the `tlsEnable` to `true`.

2. When users set both fields:
2.1. If `pinot.controller-urls` is set first and it's HTTPS, then  `pinot.tls.enabled` is already true. And user should still set`pinot.tls.enabled=true`, otherwise the validation will raise an exception.
2.2. If `pinot.tls.enabled=true` is set first, then if `pinot.controller-urls` is set HTTPS, it works, otherwise, we can still throw exception.

So the change can be:
In controller-uri set method, try to get the scheme and only set `tlsEnable` to `true` when the scheme is HTTPS.

Reply:
Changed tlsEnable from boolean to Boolean to simplify the code.